File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    
    dic = dict()
    for cloth, kind in clothes:
        if kind not in dic:
            dic[kind] = 1
        else:
            dic[kind] += 1
    
    for k in dic.values():
        answer *= k + 1
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.HashMap;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        
        HashMap<String, Integer> m = new HashMap<>();
        
        for(int i = 0; i < clothes.length; i++)
            m.put(clothes[i][1], m.getOrDefault(clothes[i][1], 0) + 1);
        
        for(String i : m.keySet())
            answer *= (m.get(i) + 1);
        
        
        return answer - 1;
    }
}